peps/pep-0781.rst

```diff
@@ -0,0 +1,280 @@
+
+There are several bindings to Zstandard for Python available on PyPI, each with
+different APIs and choices of how to bind the ``zstd`` library. One goal with
+introducing an official module in the standard library is to reduce confusion
+for Python users who want simple compression/decompression APIs for Zstandard.
+The existing packages can continue providing extended APIs and bindings for
+other Python implementations such as PyPy or integrate features from newer
+Zstandard versions.
+
+Another reason to add Zstandard support to the standard library is to resolve
+a long standing `open issue <https://github.com/python/cpython/issues/81276>`_
+requesting Zstandard support in the ``tarfile`` module. This issue has the 5th
+most "thumbs up" of open issues on the CPython tracker, and has garnered a
+significant amount of discussion and interest. The reference implementation for
+this PEP contains integration with ``tarfile`` and would address this issue.
+
+Zstandard compression could also be used to make Python wheel packages smaller
+and significantly faster to install. Anaconda found a sizeable speedup when
+adopting Zstandard for the conda package format
+
+.. epigraph::
+
+    Conda's download sizes are reduced ~30-40%, and extraction is dramatically faster.
+    [...]
+    We see approximately a 2.5x overall speedup, almost all thanks to the dramatically faster extraction speed of the zstd compression used in the new file format.
+
+    -- `Anaconda blog on Zstandard adoption <https://www.anaconda.com/blog/how-we-made-conda-faster-4-7>`_
+
+`According to lzbench <https://github.com/inikep/lzbench?tab=readme-ov-file#benchmarks>`_,
+a comprehensive benchmark of many different compression libraries and formats,
+Zstandard has a significantly higher compression ratio compared to wheel's
+existing zlib-based compression. While this PEP does *not* prescribe any
+changes to the wheel format or other packaging standards, having Zstandard
+bindings in the standard library would enable a future PEP to improve the user
+experience for Python wheel packages.
+
+Rationale
+=========
+
+Implementation based on ``pyzstd``
+----------------------------------
+
+The implementation for this PEP is based on the `pyzstd project <https://github.com/Rogdham/pyzstd>`_.
+This project was chosen as the code was `originally written to be upstreamed <https://github.com/python/cpython/issues/81276#issuecomment-1093824963>`_
+to CPython by Ma Lin, who also wrote the `output buffer implementation used in
+the standard library today <https://github.com/python/cpython/commit/f9bedb630e8a0b7d94e1c7e609b20dfaa2b22231>`_.
+The project has since been taken over by Rogdham and is published to PyPI. The
+APIs in ``pyzstd`` are similar to the APIs for other compression modules in the
+standard library such as ``bz2`` and ``lzma``.
+
+Changes and removals to ``pyzstd``'s APIs
+-----------------------------------------
+
+n.b. maybe this should be an appendix?
```

Comment:
I would agree; not all readers of this PEP will be interested in that

Reply:
Yeah, I'm not even sure if it is worthwhile as an appendix TBH, as it is much more focused on implementation details that should be hashed out in PR review I assume. I might just drop the section for now.

Reply:
I went ahead and removed this. I think it can be handled in issue discussion/PR review.